lib/ringsc.gi

```diff
@@ -918,27 +918,27 @@ InstallOtherMethod(IsUnit,"for finite Rings",
   IsCollsElms,[IsRing,IsScalar],0,
 function(R,e)
 local o,pow,a;
-  if not IsFinite(R) then
+  if not IsFinite(R) or One(R)=fail then
     TryNextMethod();
   fi;
-  if IsAssociative(R) and One(R) <> fail then
-    # compute powers of e until we reach one or repeat
-    o:=One(R);
-    pow:=[];
-    a:=e;
-    repeat
-      Add(pow,a);
-      a:=a*e;
-      if a=o then
-        # power is one. So previous power is inverse
-        return true;
-      fi;
-    until a in pow;
-    # repeats without hitting the one element, cannot be a unit
-    return false;
+
+  if not IsAssociative(R) then
```

Comment:
What is the difference herr?

Reply:
The difference is that the new code only runs if the ring is associative (it is probably not correct in the non-associative case).

Reply:
I don't understand? The code you replace also differentiates between associative and non-associative. E.g. there is a check `if IsAssociative(R) and One(R) <> fail then`

Reply:
I had noted that the code for ring.gi and ringsc.gi differed from my version. The one for ring.gi was definitively wrong. Thus I replaced both. 
I'll take the ringsc.gi out.